data/datasets.py

```diff
@@ -228,26 +224,222 @@ def __len__(self):
+            self.real_samples_paths = self.yaml_load(real_file_list_path)
+        else:
+            raise ValueError("Unknown file list type in {}".format(file_list_path))
+
+        if not no_check:
+            self.check_samples()
+        self.file_list_path = str(file_list_path)
+        self.real_file_list_path = str(real_file_list_path)
+        self.transform = transform
+        self.opts = opts
+
+    def check_samples(self):
+        """Checks that every file listed in samples_paths actually
+        exist on the file-system
+        """
+        for s in self.samples_paths:
+            for k, v in s.items():
+                assert Path(v).exists(), f"{k} {v} does not exist"
+
+        for s in self.real_samples_paths:
+            for k, v in s.items():
+                assert Path(v).exists(), f"{k} {v} does not exist"
+
+    def json_load(self, file_path):
+        with open(file_path, "r") as f:
+            return json.load(f)
+
+    def yaml_load(self, file_path):
+        with open(file_path, "r") as f:
+            return yaml.safe_load(f)
+
+    def __getitem__(self, i):
+        """Return an item in the dataset with fields:
+        {
+            data: transform({),
+            paths: [{task: path}],
+            mode: [train|val]
+        }
+        Args:
+            i (int): index of item to retrieve
+        Returns:
+            dict: dataset item where tensors of data are in item["data"] which is a dict
+                  {task: tensor}
+        """
+
+        real_i = i % len(self.real_samples_paths) - 1
+        paths = self.samples_paths[i]
+        paths["rx"] = self.real_samples_paths[real_i]["x"]
+        paths["rm"] = self.real_samples_paths[real_i]["m"]
+        paths["rd"] = self.real_samples_paths[real_i]["d"]
+
+        if self.transform:
+            # first perform transform on input images and then convert depth image
+            # to depth array
+            datadict = self.transform(
+                {task: pil_image_loader(path, task) for task, path in paths.items()}
+            )
+            datadict["d"] = get_normalized_depth(
+                datadict["d"], mode=self.opts.data.depth.sim_mode
+            )
+            datadict["rd"] = get_normalized_depth(
+                datadict["rd"], mode=self.opts.data.depth.real_mode
+            )
+            return Dict({"data": datadict, "paths": paths})
+        datadict = self.transform(
+            {task: pil_image_loader(path, task) for task, path in paths.items()}
         )
+        datadict["d"] = get_normalized_depth(
+            np.array(datadict["d"]), mode=self.opts.depth.sim_mode
+        )
+        datadict["rd"] = get_normalized_depth(
+            np.array(datadict["rd"]), mode=self.opts.depth.real_mode
+        )
+
+        return Dict({"data": datadict, "paths": paths})
+
+    def __len__(self):
+        return len(self.samples_paths)
+
+
+class SimDepthDataset(Dataset):
```

Comment:
What is so different between the `RealSimDepth` and `SimDepth` datasets that we need to have almost duplicate code? Can't we just use a few `if`s? 

I know it's the cas for non-depth datasets, maybe that should change too. It's very error-prone IMHO

Reply:
Yes I was tempted not to put any SimDepth dataset as I noticed even the rest of the code for non depth datasets is actually  not meant to be used without real data also the dataset without real exists,... 
I'll see what I can do about this